src/motile_plugin/data_views/views_coordinator/collection_widget.py

```diff
@@ -0,0 +1,406 @@
+            self.remove_track_btn.setEnabled(False)
+            self.remove_lineage_btn.setEnabled(False)
+            self.select_btn.setEnabled(False)
+
+        if selected:
+            self.collection_list.itemWidget(
+                selected[0]
+            ).update_node_count()  # update the node count
+
+            if len(self.selected_collection.collection) > 0:
+                self.select_btn.setEnabled(True)
+            else:
+                self.select_btn.setEnabled(False)
+
+        if len(self.tracks_viewer.selected_nodes) > 0:
+            self.deselect_btn.setEnabled(True)
+        else:
+            self.deselect_btn.setEnabled(False)
+
+        if self.tracks_viewer.tracks is not None:
+            self.new_group_button.setEnabled(True)
+        else:
+            self.new_group_button.setEnabled(False)
+
+    def _select_nodes(self) -> None:
+        """Select all nodes in the collection"""
+
+        selected = self.collection_list.selectedItems()
+        if selected:
+            self.selected_collection = self.collection_list.itemWidget(selected[0])
+            self.tracks_viewer.selected_nodes.add_list(
+                self.selected_collection.collection._list, append=False
+            )
+
+    def retrieve_existing_groups(self) -> None:
+        """Create collections based on the node attributes. Nodes assigned to a group should have that group in their 'group' attribute"""
+
+        # first clear the entire list
+        self.collection_list.clear()
+
+        # check for existing groups in the node attributes
+        group_dict = {}
+        for node, data in self.tracks_viewer.tracks.graph.nodes(data=True):
+            groups = data.get("group")
+            if groups:  # Only add if 'group' attribute is present and not None
+                for group in groups:
+                    if group not in group_dict:
+                        group_dict[group] = []
+                        self.add_group(name=group, select=False)
+                    group_dict[group].append(node)
+
+        # populate the lists based on the nodes that were assigned to the different groups
+        for i in range(self.collection_list.count()):
+            self.collection_list.setCurrentRow(i)
+            self.selected_collection.collection.add(
+                group_dict[self.selected_collection.name.text()]
+            )
+            self.selected_collection.update_node_count()  # enforce updating the node count for all elements
+
+        self._update_buttons_and_node_count()
+
+    def _selection_changed(self) -> None:
+        """Update the currently selected collection and send update signal"""
+
+        selected = self.collection_list.selectedItems()
+        if selected:
+            self.selected_collection = self.collection_list.itemWidget(selected[0])
+            self.group_changed.emit()
+
+        self._update_buttons_and_node_count()
+
+    def add_nodes(self, nodes: list[Any] | None = None) -> None:
```

Comment:
Call this function in the other functions (add_tracks, add_lineage) to avoid code duplication and make it easier to maintain and change